account_invoice_inter_company/tests/test_inter_company_invoice.py

```diff
@@ -153,3 +150,50 @@ def test_confirm_invoice_with_child_partner(self):
             ('auto_invoice_id', '=', self.invoice_company_a.id)
         ])
         self.assertEqual(len(invoices), 1)
+
+    def test_confirm_invoice_with_product_and_shared_catalog(self):
+        """ With no security rule, child company have access to any product.
+            Then child invoice can share the same product
+        """
+        # ensure the catalog is shared even if product is in other company
+        self.env.ref('product.product_comp_rule').write({'active': False})
+        # Product is set to a specific company
+        self.product_a.write({"company_id": self.company_a.id})
+        invoices = self._confirm_invoice_with_product()
+        self.assertNotEqual(
+            invoices.invoice_line_ids[0].product_id, False)
```

Comment:
An empty record will allways be != False, isn't it?

Reply:
you're right. Are you ok with the rest of the PR ?

Reply:
Looks ok. Anyway, fix the test to check the correct assertion

Reply:
Now fixed thanks for your review